lxd/device/config/devices.go

```diff
@@ -47,6 +48,10 @@ func (device Device) Validate(rules map[string]func(value string) error) error {
 			continue
 		}
 
+		if (k == "ceph.cluster_name" || k == "ceph.user_name") && (strings.HasPrefix(device["source"], "ceph")) {
```

Comment:
Wrong location of this, `devices.go` in most cases should not have any knowledge about the various devices.

Reply:
`rules` in disk.go is where the extra keys should go with extra validation done in `validateConfig`.

You'll also need a commit which documents the added logic in `doc/containers.md`

Reply:
I just changed the commit to reflect this. I moved the one line from the test commit (our mistake -- thanks for catching that!) and removed the logic from devices.go to the rules map in disk.go, with a check that all the necessary fields are filled.